iothub/device/src/IotHubClientOptions.cs

```diff
@@ -89,6 +89,12 @@ public IotHubClientOptions(IotHubClientTransportSettings transportSettings)
         /// <summary>
         /// Stores custom product information that will be appended to the user agent string that is sent to IoT hub.
         /// </summary>
-        public string ProductInfo { get; set; }
+        public string CustomProductInfo
+        { 
+            get => ProductInfo.Extra;
+            set => ProductInfo.Extra = value;
+        }
+
+        internal ProductInfo ProductInfo { get; } = new ProductInfo();
```

Comment:
This internal property is required because the public one stores only the extra custom text that is appended to the SDK defined user-agent string (ProductInfo).